### PR TITLE
feat: Working docker-compose healthchecks

### LINKED ISSUE #1133: Invalid Docker Compose Healthcheck
The `docker-compose.yaml` defines a fairly reasonable health check based on the SpringBoot management endpoint, running on port 7979. The `MANAGEMENT_PORT` environment variable is also set to 7979, which `application.properties` uses to control the management port used by the server.

The `docker-compose.yaml` also defines `SPRING_PROFILES_ACTIVE=dev` making the server use `application-dev.properties`, which [hardcodes the management port to 8000](https://github.com/zalando/zally/blob/master/server/src/main/resources/application-dev.yml#L27). It's not clear to me why the dev profile should force a particular management port, but there were clearly many attempts to make that the case in the early days.

As a result, when running via the default docker-compose configuration, the health check can never succeed and so the server is always marked "unhealthy".

I'm very tempted to "fix" this by dropping all reference to the management port from the dev profile... if anybody can explain why this is a bad idea then please do so!

(Obviously, another "fix" option is to remove reference to health checks from the docker-compose configuration. This seems like a backwards step to me, but am open to persuasion.)

### PR DESCRIPTION
**Pull request by roxspring**
Listing docker compose containers now shows healthy state:
```
      Name                    Command                  State                    Ports              
---------------------------------------------------------------------------------------------------
zally_postgres_1   docker-entrypoint.sh postgres    Up (healthy)   0.0.0.0:54321->5432/tcp         
zally_server_1     /bin/sh -c java $(java-dyn ...   Up (healthy)   0.0.0.0:8000->8000/tcp, 8080/tcp
zally_web-ui_1     node server.js                   Up (healthy)   0.0.0.0:8080->3000/tcp          
```

Closes #1133

### CODE REVIEW

**Comment by tkrop** [comment]
:+1:

**Comment by roxspring** [comment]
(I have noticed the CI failure btw - will try and get back to this soon if nobody beats me to it!)

**Comment by vadeg** [comment]
👍

File: server/zally-server/src/main/resources/application.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ spring:
             non_contextual_creation: true
   oauth2:
     resource:
-      tokenInfoUri: ${TOKEN_INFO_URI}
+      tokenInfoUri: ${TOKEN_INFO_URI:https://auth.example.com/oauth2/tokeninfo}
```
Comment (vadeg) [inline server/zally-server/src/main/resources/application.yml:21]: I don't think it is safe to point to some auth service by default. Users can be just unaware that it must be configured and will send their data to unknown service.

Comment (roxspring) [reply server/zally-server/src/main/resources/application.yml]: Fair point. Switched to just defaulting `MANAGEMENT_PORT`.

**Comment by vadeg** [comment]
👍

**Comment by zeitlinger** [comment]
:+1:

**Pull request merged**